script.js

```diff
@@ -1,4 +1,183 @@
+    [arr[index1], arr[index2]] = [arr[index2], arr[index1]];
+  };
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr.length - i; j++) {
+      if (arr[j] < arr[j + 1] && mode == "big") {
+        swap(arr, j, j + 1);
+      } else if (arr[j] > arr[j + 1] && mode == "small") {
+        swap(arr, j, j + 1);
+      }
+    }
+  }
+  return arr;
+}
+
+// Get the biggest/lowest dice number combination
+function getDiceCombo(num_of_dice, mode) {
+  let counter = 0;
+  let arr = [];
+  while (counter < Number(num_of_dice)) {
+    arr.push(rollDice());
+    counter += 1;
+  }
+  numsArr = arr.toString();
+  let sortedArr = sortDiceNumArr(arr, mode);
+  let numStr = "";
+  for (let i = 0; i < sortedArr.length; i++) {
+    numStr += String(sortedArr[i]);
+  }
+  let combination = Number(numStr);
+  return combination;
+}
+
+// Generate a sorted leaderboard to show each player's optimal number
+function generateLeaderboard(arr, mode) {
+  let record;
+  let showRecord = "";
+  let leaderboard = [];
+  for (let i = 0; i < arr.length; i++) {
+    let player = `Player ${i + 1}`;
+    leaderboard.push({ player });
+    leaderboard[i].value = arr[i];
+  }
+  if (mode == "big") {
+    record = leaderboard.sort((a, b) => (a.value < b.value ? 1 : -1));
+  } else {
+    record = leaderboard.sort((a, b) => (a.value > b.value ? 1 : -1));
+  }
+  for (let j = 0; j < record.length; j++) {
+    showRecord +=
+      `${[j + 1]}. ` + record[j].player + " : " + record[j].value + `<br>`;
+  }
+  return showRecord;
+}
+
+// Find the index of duplicated number in game board array
+function findDuplicatesIndex(arr, num) {
+  let duplicatesIndex = [];
+  let idx = arr.indexOf(num);
+  while (idx != -1) {
+    duplicatesIndex.push(idx);
+    idx = arr.indexOf(num, idx + 1);
+  }
+  return duplicatesIndex;
+}
+
+// Check winner and display
+function checkWinner(numOfDice, numOfPlayer, gameMode) {
+  if (playerCounter <= numOfPlayer) {
+    gameRecordArr.push(getDiceCombo(numOfDice, gameMode));
+    for (let i = 0; i < gameRecordArr.length; i++) {
+      let diceNumsStr = "";
+      for (let j = 0; j < numsArr.length; j++) {
+        diceNumsStr += " " + numsArr[j] + " ";
+      }
+      // Display each player's dice number and optimal number combo
+      diceRollMsg = `Player ${playerCounter}, you roll "${diceNumsStr}", your number is ${gameRecordArr[i]}<hr>`;
+    }
+
+    gameBoard = generateLeaderboard(gameRecordArr, gameMode);
+    playerCounter++;
+    submitBtn.innerHTML = `Player ${playerCounter}`;
+
+    if (playerCounter > numOfPlayer) {
+      let maxNum = Math.max(...gameRecordArr);
+      let minNum = Math.min(...gameRecordArr);
+
+      // Find the index of maxNum if have more then two maxNum
+      let dulNum =
+        gameMode == "big"
+          ? Math.max(...gameRecordArr)
+          : Math.min(...gameRecordArr);
+      let duplicatedNumIndexArr = findDuplicatesIndex(gameRecordArr, dulNum);
+      let dulTopPlayers = "";
```

Comment:
why not 

```
let dulNum = gameMode == 'big' ? maxNum : minNum
```

seems a bit redundant to repeat the Math.max and Math.min calculation